src/components/VisitInput.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { ChangeEvent, useState } from "react";
+import { Button, Stack, TextField } from "@mui/material";
+
+interface Visit {
+  proposalCode: string;
+  proposalNumber: number;
+  number: number;
+}
+
+interface VisitInputProps {
+  onSubmit: (visit: Visit, parameters?: object) => void;
+  visit?: Visit;
+  parameters?: object;
+}
+
+const VisitInput: React.FC<VisitInputProps> = ({
+  onSubmit,
+  visit,
+  parameters,
+}) => {
+  const [visitText, setVisitText] = useState(
+    visit
+      ? `${visit.proposalCode}${visit.proposalNumber.toFixed(
+          0
+        )}-${visit.number.toFixed(0)}`
+      : ""
+  );
+  const [isValid, setIsValid] = useState(true);
+
+  const visitRegex = /^([a-z]{2})([1-9]\d*)-([1-9]\d*)/;
```

Comment:
Can we have a test for this regex. Something that just tests a group of valid and invalid ones.

Reply:
I've added tests and also modified the regex slightly to prvent any extra trailing groups.